### PR TITLE
Autocomplete

### PR DESCRIPTION
**Pull request by JBigalke**
Added autocomplete functions for editMom.
For attribute "id" in the element persName a service is called to create a json from the personlist with the matching string. 
- The user can search for ids and for names but, always the id of the person is stored.
- personlists from the database are only called when required and the json is stored in a session-variable

### CODE REVIEW

File: my/XRX/src/mom/app/collection/service/my-collection-charter-edit-get-personlist-as-json.service.xml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+     {
+        (:
+            Check if a new search is required. For this check if a json in session-variables exists. if the searchmode
+        has changed or the searchterm has changed.
+        :)
+        let $newsearch :=
+            if(not(empty($beforemode)) and not(empty($beforeterm)) and not(empty($beforelist))) then(
+                if ($mode != $beforemode or $searchterm != $beforeterm)
+            then(
+                if($searchterm gt $beforeterm) then(
```
Comment (StephanMa) [inline my/XRX/src/mom/app/collection/service/my-collection-charter-edit-get-personlist-as-json.service.xml:76]: Whats your intention with this line? Do you want to compare the length or the content?

Comment (JBigalke) [reply my/XRX/src/mom/app/collection/service/my-collection-charter-edit-get-personlist-as-json.service.xml:76]: I want to compare the length it should be
  if(string-length($searchterm) gt string-length($beforeterm)) then(

Comment (JBigalke) [reply my/XRX/src/mom/app/collection/service/my-collection-charter-edit-get-personlist-as-json.service.xml]: Fix is commited

**Comment by StephanMa** [comment]
Does not work

```
Function xmleditor:personlistautocompletejson() is not defined in namespace 'http://www.monasterium.net/NS/xmleditor' [at line 161, column 26] In function: local:eval(element()) [292:30:/db/XRX.live/mom/app/xrx/main.xql] local:mode-mainwidget() [429:13:/db/XRX.live/mom/app/xrx/main.xql] local:main() [446:1:/db/XRX.live/mom/app/xrx/main.xql]
```

**Comment by StephanMa** [comment]
I can compile the app now.
Can yoou give me a hint where to test this?
I modified a charter, added a persName... and after this?

**Comment by JBigalke** [comment]
persName --> id and then into the textfield in for example P_Angel for an autocomplete over id or Johan for autocomplete over name.

**Comment by StephanMa** [comment]
Unbenannt
Nothing happens here

**Comment by JBigalke** [comment]
> Unbenannt
> Nothing happens here

It seems there is a problem with the different Languages, In the English Mom version it works.. in the German not. And I only tested it in the English version. 
I try to fix this quick

**Comment by JBigalke** [comment]
I changed the test to get the right Element now it should work for all languages

**Comment by StephanMa** [comment]
Works.... merged

**Comment by GVogeler** [comment]
@JBigalke did you really link the selection to the `@id` field? This should be `@key` as `@id` is an identifier in the local document (`@xml:id`), as you might remember from our mail exchanges :-) . Can you change this before deployment, please?
I've tried to describe the use case in #950

**Comment by StephanMa** [comment]
Lucky.. today is a workshop in Naples so I don’t want to deploy stuff today...

**Comment by JBigalke** [comment]
@GVogeler I changed it:

grafik


grafik

**Comment by GVogeler** [comment]
btw: this PR should solve #950

**Pull request merged**